src/header.iced

```diff
@@ -14,50 +13,60 @@ crypto_auth_KEYBYTES = 32
 crypto_secretbox_KEYBYTES = 32
 
 compute_mac_key = (encryptor, header_hash, pubkey) ->
-  zero_bytes = Buffer.alloc(crypto_auth_KEYBYTES)
+  zero_bytes = new Buffer(crypto_auth_KEYBYTES)
+  zero_bytes.fill(0)
   mac_box = encryptor.encrypt({plaintext : zero_bytes, nonce : nonce.nonceForMACKeyBox(header_hash), pubkey})
   # take last crypto_auth_BYTES bytes of MAC box
   return mac_box.slice(-crypto_auth_KEYBYTES)
 
-exports.generate_encryption_header_packet = (encryptor, recipients) ->
+exports.generate_encryption_header_packet = (encryptor, recipients, opts) ->
+  # create the header list and populate the quick stuff
   header_list = []
   header_list.push('saltpack')
   header_list.push([current_major, current_minor])
   header_list.push(encryption_mode)
 
-  payload_encryptor = crypto.alloc({force_js : true})
-  payload_key = prng(crypto_secretbox_KEYBYTES)
+  # generate the payload and ephemeral keys
+  payload_encryptor = nacl.alloc({force_js : false})
+  payload_key = crypto.randomBytes(crypto_secretbox_KEYBYTES)
```

Comment:
Do we know what `randomBytes` is backed by? (On all the different versions of node people might still be using in production?) I've seen [big threads](https://github.com/nodejs/node/issues/5798) arguing about this, and I don't know what they decided.